### PR TITLE
ISPN-8022 Address Radim's comments WRT Hibernate Cache Provider

### PR DESCRIPTION
**Pull request by galderz**
https://issues.jboss.org/browse/ISPN-8022

Address remaining comments from @rvansa on #5254 and #5268 PRs.

### CODE REVIEW

File: hibernate/cache-commons/src/main/java/org/infinispan/hibernate/cache/commons/access/LockingInterceptor.java [hunk 1]
```diff
@@ -34,18 +35,12 @@
...
-            try {
-               ((CompletableFuture) rv).join();
-            }
-            catch (CompletionException e) {
-               throw e.getCause();
-            }
-         }
+   protected final InvocationFinallyFunction unlockAllReturnCheckCompletableFutureHandler = (rCtx, rCommand, rv, throwable) -> {
+      lockManager.unlockAll(rCtx);
+      if (rv instanceof CompletableFuture) {
```
Comment (rvansa) [inline hibernate/cache-commons/src/main/java/org/infinispan/hibernate/cache/commons/access/LockingInterceptor.java]: If `throwable != null` you should rethrow.

**Comment by galderz** [comment]
@rvansa Fixed comment

**Comment by rvansa** [comment]
@galderz Ok, will you squash this or should I integrate this as-is when CI gives me green?

**Comment by galderz** [comment]
Let me squash

**Comment by galderz** [comment]
@rvansa Squashed

**Comment by rvansa** [comment]
Integrated, thanks!

**Pull request merged**